From a2effe1c3244fbf28a4f413ad409f51a7e8f47a8 Mon Sep 17 00:00:00 2001
From: Mark Janes <mark.a.janes@intel.com>
Date: Fri, 29 Apr 2016 12:15:05 -0700
Subject: [PATCH] Attempt to load prebuilt spirv from cache.

We save test execution time by prebuilding spirv.  The cts should only
compile spirv at run time if it fails to find the shader in the cache.
---
 external/vulkancts/modules/vulkan/vktTestPackage.cpp | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/external/vulkancts/modules/vulkan/vktTestPackage.cpp b/external/vulkancts/modules/vulkan/vktTestPackage.cpp
index 2d85aad..18664c4 100644
--- a/external/vulkancts/modules/vulkan/vktTestPackage.cpp
+++ b/external/vulkancts/modules/vulkan/vktTestPackage.cpp
@@ -92,18 +92,14 @@ vk::ProgramBinary* buildProgram (const std::string&					casePath,
 
 	try
 	{
-		binProg	= de::MovePtr<vk::ProgramBinary>(compileProgram(iter.getProgram(), &buildInfo));
-		log << buildInfo;
-	}
-	catch (const tcu::NotSupportedError& err)
-	{
-		// Try to load from cache
-		log << err << tcu::TestLog::Message << "Building from source not supported, loading stored binary instead" << tcu::TestLog::EndMessage;
-
 		binProg = de::MovePtr<vk::ProgramBinary>(prebuiltBinRegistry.loadProgram(progId));
-
 		log << iter.getProgram();
 	}
+    catch (const vk::ProgramNotFoundException &err)
+    {
+        binProg = de::MovePtr<vk::ProgramBinary>(compileProgram(iter.getProgram(), &buildInfo));
+		log << buildInfo;
+    }
 	catch (const tcu::Exception&)
 	{
 		// Build failed for other reason
-- 
2.8.0.rc3

